src/core/libmaven/isotopeDetection.cpp

```diff
@@ -164,58 +152,83 @@ map<string, PeakGroup> IsotopeDetection::getIsotopes(PeakGroup* parentgroup, vec
+    {
+        float expectedAbundance = x.abundance;
+        if (expectedAbundance < 1e-8)
+            return true;
+        if (expectedAbundance * parentPeakIntensity < 1) //TODO: In practice this is probably fine but in general I don't like these types of intensity checks -- the actual absolute value depends on the type of instrument, etc
+            return true;
+        float observedAbundance = isotopePeakIntensity
+            / (parentPeakIntensity + isotopePeakIntensity); //find observedAbundance based on isotopePeakIntensity
+
+        float naturalAbundanceError = abs(
+                observedAbundance - expectedAbundance) //if observedAbundance is significant wrt expectedAbundance
+            / expectedAbundance * 100; // compute natural Abundance Error
+
+        if (naturalAbundanceError >
+                _mavenParameters->maxNaturalAbundanceErr)
+            return true;
+    }
+
+    //TODO: this is really an abuse of the maxIsotopeScanDiff parameter
+    //I can easily imagine you might set maxIsotopeScanDiff to something much less than the peak width
+    //here w should really be determined by the minRt and maxRt for the parent and child peaks
+    if (parentGroup)
     {
-        //TODO: this is really an abuse of the maxIsotopeScanDiff parameter
-        //I can easily imagine you might set maxIsotopeScanDiff to something much less than the peak width
-        //here w should really be determined by the minRt and maxRt for the parent and child peaks
-        float w = _mavenParameters->maxIsotopeScanDiff
-            * _mavenParameters->avgScanTime;
-        float isotopeMass = x.mass;
         Peak* parentPeak = parentGroup->getPeak(sample);
-        float rtmin = parentPeak->rtmin;
-        float rtmax = parentPeak->rtmax;
+        float rtmin = parentGroup->minRt;
+        float rtmax = parentGroup->maxRt;
+        if (parentPeak)
+        {
+            rtmin = parentPeak->rtmin;
+            rtmax = parentPeak->rtmax;
+        }
+        float isotopeMass = x.mass;
         float parentMass = parentGroup->meanMz;
+        float w = _mavenParameters->maxIsotopeScanDiff
+            * _mavenParameters->avgScanTime;
         double c = sample->correlation(
                 isotopeMass, parentMass,
                 _mavenParameters->compoundMassCutoffWindow, rtmin - w,
                 rtmax + w, _mavenParameters->eicType,
-                _mavenParameters->filterline);  // find correlation for isotopes
+            _mavenParameters->filterline);  // find correlation for isotopes
         if (c < _mavenParameters->minIsotopicCorrelation)
             return true;
     }
     return false;
 }
 
+std::pair<float, float> IsotopeDetection::getIntensity(Scan* scan, float mzmin, float mzmax)
+{
+    float highestIntensity = 0;
+    float rt = 0;
+    mzSample* sample = scan->getSample();
+    //TODO: use maxIsotopeScanDiff instead of arbitrary number
+    for (int i = scan->scannum - 2; i < scan->scannum + 2; i++) {
```

Comment:
@shubhra-agrawal 
it's not clear as to why we are starting from scan number - 2 and end at scan number + 2. 
Document this in the comments, no  ?

Reply:
@rish9511 That's old code. To make sure isotopic peak is within 2 scans of the parent. We have a variable called maxIsotopeScanDiff which should be used here instead but it was messing up the results a bit so I changed it back. I've added a TODO and will take care of this when possible. 
For now, I'll add a comment if you think that's useful.

Reply:
@shubhra-agrawal 
let's leave it then